### PR TITLE
WIP - Break up utils.py

### PR DESCRIPTION
**Pull request by martica**
This is an attempt to break out the parts of utils.py. 

## PR Checklist:

- [ ] All new features have been tested
- [ ] All new features have been documented
- [ ] I have read the **CONTRIBUTING.md** file
- [ ] I will abide by the code of conduct

### CODE REVIEW

File: tests/utils/transpile_test_case.py [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+        line1, val1 = _normalize(line1)
+        line2, val2 = _normalize(line2)
+        if transform_output(val1) == transform_output(val2):
+            line2 = line1
+        elif (
+            type(val1) == type(val2) and
+            type(val1) in (float, complex) and
+            val1 + val2 != 0 and
+            abs(val1 - val2) / abs(val1 + val2) < 0.0001):
+            line2 = line1
```
Comment (brutusthebee) [inline tests/utils/transpile_test_case.py]: At column 13: (E125) continuation line with same indent as next logical line

File: tests/utils/output_cleaners.py [hunk 2]
```diff
@@ -0,0 +1,234 @@
+import os
+import re
+import sre_constants
+
+JS_EXCEPTION = re.compile(
+    'Traceback \(most recent call last\):\r?\n(  File "(?P<file>.*)", line (?P<line>\d+), in .*\r?\n)+(?P<exception>.*?): (?P<message>.*\r?\n)')  # NOQA
```
Comment (brutusthebee) [inline tests/utils/output_cleaners.py]: At column 12: (W605) invalid escape sequence '\('

File: tests/utils/output_cleaners.py [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+            'decimal': decimal,
+            'float_exp': float_exp,
+            'complex_num': complex_num,
+            'high_precision_float': high_precision_float,
+            'test_ref': test_ref,
+            'custom': custom
+        }
+
+    def cleanse(self, js_input, substitutions):
+        """
```
Comment (brutusthebee) [inline tests/utils/output_cleaners.py]: At column 5: (C901) 'JSCleaner.cleanse' is too complex (18)

File: tests/utils/output_cleaners.py [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+            'memory_ref': memory_ref,
+            'float_exp': float_exp,
+            'complex_num': complex_num,
+            'high_precision_float': high_precision_float,
+            'test_ref': test_ref,
+            'custom': custom
+        }
+
+    def cleanse(self, py_input, substitutions):
+        """
```
Comment (brutusthebee) [inline tests/utils/output_cleaners.py]: At column 5: (C901) 'PYCleaner.cleanse' is too complex (13)

**Comment by martica** [comment]
His complexity complaints are useful, at least.

**Pull request merged**